tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/TelemetryHelper.cs

```diff
@@ -165,9 +165,45 @@ _ when data.TryGetPayload<AppIntegrationsChangedPayload>(TelemetryRequestTypes.A
                 }
             }
 
+            var spansCreatedByIntegration = new ConcurrentDictionary<string, MetricData>();
+            var metricsPayloads =
+                allData
+                    .Select(
+                        data => data switch
+                        {
+                            _ when data.TryGetPayload<GenerateMetricsPayload>(TelemetryRequestTypes.GenerateMetrics) is { } p => p.Series.Where(s => s.Metric == "spans_created"),
+                            _ => null,
+                        })
+                    .Where(x => x is not null);
+
+            // Flatten the spans_created metrics
+            foreach (var metricPayload in metricsPayloads)
+            {
+                foreach (var metricEntry in metricPayload)
+                {
+                    spansCreatedByIntegration.TryAdd(metricEntry.Tags.First(s => s.StartsWith("integration_name:")), metricEntry);
+                }
+            }
+
+            var integrationName = integrationId.ToString();
+            var integrationNameTagValue = integrationName switch
+            {
+                "OpenTelemetry" => "otel",
+                _ => integrationName.ToLowerInvariant(),
+            };
```

Comment:
Not sure if you _want_ to do this or not, because this grabs directly from the metric tag:

```suggestion
            var integrationNameTagValue = integrationId.GetMetricTag()
```

Reply:
I was thinking of not doing this, so that we could write the assertion without invoking the exact product code. So this `"otel"` or convert from integration name seemed like a better test case